tasks/tag_repo_on_crate_version_change.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+
+// # Overview
+//
+// Automatically tags the repo with a "version tag" when the version of the repo's
+// crate or specified crate changes when merged to main. For example, say you have
+// version 1.1.0 tagged and you merge in a commit to main that changes the version in
+// Cargo.toml to 1.1.1... this would detect that and tag the repo with 1.1.1.
+//
+// Note: this will detect whether to add a `v` prefix or not based on what the
+// repo mostly does.
+//
+// # Example Use
+//
+// ```yml
+// - name: Tag Version on Change
+//   if: |
+//     github.repository == 'denoland/<REPO_NAME_GOES_HERE>' &&
+//     github.ref == 'refs/heads/main'
+//    run: deno run --allow-read --allow-run=cargo,git <url-to-this-module>
+// ```
+//
+// If you have multiple crates, then you will need to provide the crate name
+// to use to determine the tag name as an argument to the script.
+//
+// ```bash
+// deno run --allow-read --allow-run=cargo,git <url-to-this-module> <crate-name-goes-here>
+// ```
```

Comment:
I think it would be good if this also automatically created a github release draft with the git log since the last tag. I will look into it tomorrow.

https://docs.github.com/en/rest/reference/releases#create-a-release